test/new-e2e/tests/containers/ecs_test.go

```diff
@@ -77,7 +78,12 @@ func (suite *ecsSuite) SetupSuite() {
 	suite.Require().NoError(fakeintake.Init(suite))
 	suite.Fakeintake = fakeintake.Client()
 
-	suite.ecsClusterName = stackOutput.Outputs["ecs-cluster-name"].Value.(string)
+	clusterSerialized, err := json.Marshal(stackOutput.Outputs["dd-Cluster-ecs-cluster"].Value)
+	suite.Require().NoError(err)
+	ecsCluster := &ecsComp.ClusterOutput{}
+	suite.Require().NoError(ecsCluster.Import(clusterSerialized, ecsCluster))
+
+	suite.ecsClusterName = ecsCluster.ClusterName
```

Comment:
This looks as if we were JSON-marshaling a structured data line 81 to unmarshal it line 84.

Wasn’t it possible to extract cluster name directly from the original structured data without having to serialize and deserialize it ?

Reply:
I considered it initially, it requires few more lines of code to extract and cast the items from the map. This is a temporary fix, as @KevinFairise2 is working on migrating this test to the provisioner based suite. I can change it back to the extraction from the map to improve performance - indeed extracting + casting would be a 0(1) operation.